core/src/main/java/org/apache/druid/common/config/ConfigManager.java

```diff
@@ -168,64 +171,99 @@ private void poll()
     return holder.getReference();
   }
 
-
   public <T> SetResult set(final String key, final ConfigSerde<T> serde, final T obj)
   {
-    if (obj == null || !started) {
-      if (obj == null) {
-        return SetResult.fail(new IllegalAccessException("input obj is null"));
+    return set(key, serde, null, obj);
+  }
+
+  public <T> SetResult set(final String key, final ConfigSerde<T> serde, @Nullable final T oldObject, final T newObject)
+  {
+    if (newObject == null || !started) {
+      if (newObject == null) {
+        return SetResult.fail(new IllegalAccessException("input obj is null"), false);
```

Comment:
super nitpick: i know this isn't new or your fault, but "obj" doesn't seem to need to be shortened